docker/Dockerfile.cpu

```diff
@@ -0,0 +1,56 @@
+##############################################################
+# This Dockerfile contains the Devito codes and can be buld using different base images.
+##############################################################
+
+ARG arch=gcc
+
+# Base image 
+FROM python:3.7 as base
+
+ENV DEBIAN_FRONTEND noninteractive
+
+# Install for basic base not containing it
+RUN apt-get update && apt-get install -y mpich libmpich-dev libgl1-mesa-glx
+
+##############################################################
+# GCC standard image
+##############################################################
+FROM base as gcc
+# Env vars defaults
+ENV DEVITO_ARCH="gcc"
+ENV DEVITO_LANGUAGE="openmp"
+
+##############################################################
+# ICC image
+##############################################################
+FROM base as icc
+
+ # download the key to system keyring
+RUN wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor > /usr/share/keyrings/oneapi-archive-keyring.gpg
+RUN echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" > /etc/apt/sources.list.d/oneAPI.list
+
+# Install wanted components only (icc and mpiicc)
+RUN apt-get update -y && apt-get install -y intel-oneapi-compiler-dpcpp-cpp-and-cpp-classic intel-oneapi-mpi
+
+# Set en vars by hand since we can't use entrypoint for intermediate image
+ENV ICC_HOME=/opt/intel/oneapi/compiler/latest/
+ENV MPICC_HOME=/opt/intel/oneapi/mpi/latest/
+ENV PATH $MPICC_HOME/bin:$MPICC_HOME/include:$ICC_HOME/linux/bin/intel64:$ICC_HOME/libfabric/bin:${PATH}
+ENV LD_LIBRARY_PATH $ICC_HOME/linux/lib:$ICC_HOME/linux/compiler/lib/intel64_lin:$MPICC_HOME/lib:$MPICC_HOME/include:${LD_LIBRARY_PATH}
+
+# Devito config
+ENV DEVITO_ARCH="icc"
+ENV DEVITO_LANGUAGE="openmp"
+# C compiler for mpi4py
+ENV MPICC="mpiicc"
```

Comment:
will `mpi4py` work? (@georgebisbas this might be worth a try)

Reply:
It does on my desktop yeah